src/mon/OSDMonitor.cc

```diff
@@ -6274,7 +6276,35 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
   // This should be used as a general guideline for most commands handled
   // in this function.  Adapt as you see fit, but please bear in mind that
   // this is the expected behavior.
-   
+
+  if (prefix == "osd test") {
+    // 1. make sure other paxosservices are writeable.
+    if (!mon->authmon()->is_writeable()) {
+      dout(10) << "waiting for auth mon" << dendl;
+      mon->authmon()->wait_for_writeable(op, new C_RetryMessage(this, op));
```

Comment:
If at any point we see that this op ends up being starved, we can always add it to the front of the queue instead. It may not be trivial because `wait_for_writeable()` will queue on different queues depending on why we are not writeable, but shouldn't be too hard to figure out what goes where (or even simplifying that logic).